### ISSUE TITLE
reproducing the performance of XLM-ROBERTA on MLQA dataset on the zh language

### ISSUE THREAD

**Issue by nooralahzadeh** [huggingface/transformers#3510]
# ❓ Questions & Help
## Details

I have trouble in reproducing the result of XLM-ROBERTA on MLQA dataset for Chinese languages. The result of the rest of the language seems fine, however on the zh section, I have very low  {'exact_match': 4.263188631496982, 'f1': 17.451059178461946})!

**Comment by nooralahzadeh** [comment]
Credit to people at Microsoft Asia: Ning Wu and Nan Dua
To achieve the best performance on the "zh" test sets, you just need to add 
"final_text = tok_text" after line 497 in squad_metrics.py (only for zh). Because there isn't space and subword in Chinese, so we don't need to execute the get_final_test() function.

**Comment by LysandreJik** [comment]
This is very interesting! Thanks for letting us know @nooralahzadeh!

**Comment by nooralahzadeh** [comment]
@LysandreJik  Do you think the training model  "RobertaForQuestionAnswering" also need to be updated for 'zh' lang.  Because when I try to fine-tune the xlm-r on 'zh' language and evaluate on its test set, the results became very lower than not fine-tuning.

**Comment by PluviophileYU** [comment]
Just found the same problem, thanks bro!

> Credit to people at Microsoft Asia: Ning Wu and Nan Dua
> To achieve the best performance on the "zh" test sets, you just need to add
> "final_text = tok_text" after line 497 in squad_metrics.py (only for zh). Because there isn't space and subword in Chinese, so we don't need to execute the get_final_test() function.

**Comment by tnq177** [comment]
I used huggingface to train & predict then use https://github.com/facebookresearch/MLQA/blob/main/mlqa_evaluation_v1.py to calculate the scores. Seem to match the paper.
